backend/controllers/EventParticipation.js

```diff
@@ -19,11 +20,36 @@ const invitedUserEmails = async (req, res) => {
 
 /**
  * Controller function to get all pending invitations of a user
- * Asignee: chi Robin
+ * Asignee: Jenny
  * Name: getPendingInvitations
  * req: request object containing user id.
  * res: response object to return status 200.
  */
+const getPendingInvitations = async (req, res) => {
+  try {
+    // need to change schema to get the person who send the invitation
+    const userId = req.user._id;
+    const participation = await EventParticipation.find({
+      userId: userId,
+      status: 'pending',
+    });
+
+    const result = await Promise.all(
+      participation.map(async participation => {
+        const invitor = await User.findById(participation.invitedBy);
+        let tempEvent = await Event.findById(participation.eventId);
+        tempEvent = JSON.parse(JSON.stringify(tempEvent));
+        const modifiedResult = { ...tempEvent, invitedBy: invitor.email };
```

Comment:
I am curious why you have to stringify and parse here em? It seems non-intuititve. If for any reasons that you plan to keep it that way, can you add some comments explaining the reason?

Reply:
i have replaced it with the function lean() from mongoose!